python/hsfs/core/feature_group_engine.py

```diff
@@ -38,9 +38,23 @@ def save(self, feature_group, feature_dataframe, write_options):
                 feat.primary = True
             if feat.name in feature_group.partition_key:
                 feat.partition = True
+            if (
+                feature_group.hudi_precombine_key is not None
+                and feat.name in feature_group.hudi_precombine_key
+            ):
+                feat.hudi_precombine_key = True
 
         self._feature_group_api.save(feature_group)
 
+        # if hudi precombine key was not provided and TimeTravelFormat is HUDI, retrieve from backend and set
+        if (
+            feature_group.hudi_precombine_key is None
+            and feature_group.time_travel_format == "HUDI"
+        ):
+            for feat in self.features:
+                if feat.hudi_precombine_key:
+                    feature_group.hudi_precombine_key = feat.name
+
```

Comment:
You shouldn't need this step, the Python feature_group_api automatically updates the `feature_group` object with what comes back from the backend.